screens/CreateWorkout.js

```diff
@@ -55,20 +100,31 @@ class CreateWorkoutScreen extends React.Component {
     let exercises = this.state.exercises;
     exercises.push(exercise);
     this.setState({ exercises });
+    this.setState({ saved: false });
   }
 
   changeExerciseName(name, i) {
     let exercises = this.state.exercises;
     let exercise = exercises[i];
     exercise.name = name;
     this.setState({ exercises });
+    this.setState({ saved: false });
```

Comment:
just a little thing. you change state one the prev. line, and again on this line so it will call render() twice. you can set both props ('exercises' and 'saved ' in one call)

Reply:
Good catch!